public/app/features/variables/adhoc/adapter.ts

```diff
@@ -35,5 +38,24 @@ export const createAdHocVariableAdapter = (): VariableAdapter<AdHocVariableModel
       const filters = variable?.filters ?? [];
       return urlParser.toUrl(filters);
     },
+    beforeAdding: (model) => {
+      if (!isAdHoc(model)) {
+        return model;
+      }
+
+      if (!isLegacyAdHocDataSource(model.datasource)) {
+        return model;
+      }
```

Comment:
is this necessary?  `getInstanceSettings()` should work with a string (name|uid|ref|null)

Reply:
I want to do the upgrade only if the datasource prop is a string otherwise exit fast. Just being cautious about what code we're running.